modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,345 @@
+  });
+
+  it('TINY-11974: Should return multiple promises for multiple userIds', () => {
+    const editor = hook.editor();
+    const userIds = [ 'test-user-1', 'test-user-2' ];
+
+    const promises = editor.userLookup.fetchUsers(userIds);
+
+    expect(promises).to.have.lengthOf(2, 'Should return array with two promises');
+    expect(promises[0]).to.be.instanceOf(Promise, 'First item should be a Promise');
+    expect(promises[1]).to.be.instanceOf(Promise, 'Second item should be a Promise');
+  });
+
+  it('TINY-11974: Should throw an error for empty string ids', async () => {
+    const editor = hook.editor();
+    const userIds = [ '' ];
+    const [ userPromise ] = editor.userLookup.fetchUsers(userIds);
+
+    await expect(userPromise).to.eventually.be.rejectedWith(
+      Error,
+      'User  not found',
+      'Should throw error for empty string ID'
+    );
+  });
+
+  it('TINY-11974: Should resolve with user data when fetching a single user', async () => {
+    const editor = hook.editor();
+    const userId = 'test-user-1';
+
+    const [ userPromise ] = editor.userLookup.fetchUsers([ userId ]);
+
+    const expectedUser = {
+      id: userId,
+      name: 'Test User',
+      avatar: 'test-avatar.png',
+      description: 'Test Description'
+    };
+
+    await expect(userPromise).to.eventually.deep.equal(
+      expectedUser,
+      'Should resolve with expected user object'
+    );
+  });
+
+  it('TINY-11974: Should cache and return same data for subsequent requests', async () => {
+    const editor = hook.editor();
+    const userId = 'test-user-1';
+
+    const [ firstPromise ] = editor.userLookup.fetchUsers([ userId ]);
+    const [ secondPromise ] = editor.userLookup.fetchUsers([ userId ]);
+
+    await expect(firstPromise).to.eventually.deep.equal(
+      await secondPromise,
+      'Should return same data for subsequent requests'
+    );
+  });
+
+  it('TINY-11974: Should reject promise for non-existent users', async () => {
+    const editor = hook.editor();
+    const nonExistentId = 'non-existent-user';
+
+    const [ userPromise ] = editor.userLookup.fetchUsers([ nonExistentId ]);
+
+    await expect(userPromise).to.eventually.be.rejectedWith(
+      Error,
+      `User ${nonExistentId} not found`,
+      'Should reject for non-existent user'
+    );
+  });
+
+  it('TINY-11974: Should handle repeated requests for non-existent users', async () => {
+    const editor = hook.editor();
+    const nonExistentId = 'non-existent-user';
+    const errorMessage = `User ${nonExistentId} not found`;
+
+    const [ firstPromise ] = editor.userLookup.fetchUsers([ nonExistentId ]);
+    await expect(firstPromise).to.eventually.be.rejectedWith(errorMessage);
+
+    // Second request should use cached rejection
+    const [ secondPromise ] = editor.userLookup.fetchUsers([ nonExistentId ]);
+    await expect(secondPromise).to.eventually.be.rejectedWith(errorMessage);
```

Comment:
You're not actually checking if it's a cached rejection, similar to the other cached test